File: ibkr_show_backend/app/schemas/cash_flows.py
from pydantic import BaseModel

from app.schemas.common import PaginationInfo


class CashFlowItem(BaseModel):
    account_id: str
    currency: str | None = None
    description: str | None = None
    date_time: str | None = None
    settle_date: str | None = None
    available_for_trading_date: str | None = None
    amount: float | None = None
    flow_direction: str | None = None
    flow_type: str | None = None
    transaction_id: str | None = None
    report_date: str | None = None
    client_reference: str | None = None


class CashFlowListResponse(BaseModel):
    items: list[CashFlowItem]
    pagination: PaginationInfo


class CashFlowCurrencySummaryItem(BaseModel):
    currency: str | None = None
    record_count: int
    deposit_count: int
    withdrawal_count: int
    total_deposit_amount: float
    total_withdrawal_amount: float
    net_amount: float


class CashFlowSummaryResponse(BaseModel):
    record_count: int
    deposit_count: int
    withdrawal_count: int
    total_deposit_amount: float | None = None
    total_withdrawal_amount: float | None = None
    net_amount: float | None = None
    by_currency: list[CashFlowCurrencySummaryItem]

File: ibkr_show_frontend/src/types/cashFlows.ts
import type { PaginationInfo } from './common'

export interface CashFlowItem {
  account_id: string
  currency: string | null
  description: string | null
  date_time: string | null
  settle_date: string | null
  available_for_trading_date: string | null
  amount: number | null
  flow_direction: string | null
  flow_type: string | null
  transaction_id: string | null
  report_date: string | null
  client_reference: string | null
}

export interface CashFlowListResponse {
  items: CashFlowItem[]
  pagination: PaginationInfo
}

export interface CashFlowSummaryResponse {
  record_count: number
  deposit_count: number
  withdrawal_count: number
  total_deposit_amount: number | null
  total_withdrawal_amount: number | null
  net_amount: number | null
  by_currency: CashFlowCurrencySummaryItem[]
}

export interface CashFlowCurrencySummaryItem {
  currency: string | null
  record_count: number
  deposit_count: number
  withdrawal_count: number
  total_deposit_amount: number
  total_withdrawal_amount: number
  net_amount: number
}
